build_sdk_container_image

```diff
@@ -0,0 +1,145 @@
+    echo "      -c           - Use scripts and ebuild repos as-is, do not check out version tag."
+    echo "      -d             Indicates the container to be built is based on a development/nightly"
+    echo "                     SDK version, not a released SDK"
+    echo "      -p             Publish containers to '$sdk_container_common_registry' after successful build."
+    echo "                     requires docker to be authenticated to the registry."
+    echo "      -h             Print help"
+    echo
+}
+
+# --
+
+while [ 0 -lt $# ] ; do
+    case "$1" in
+    -h) usage; exit 0;;
+    -v) version="$2"; shift; shift;;
+    -a) arch="$2";    shift; shift;;
+    -f) tarball="$2"; shift; shift;;
+    -e) existing="true";     shift;;
+    -c) checkout="false";    shift;;
+    -d) dev="true";          shift;;
+    -p) push="true";         shift;;
+    *)  usage; exit 1;;
+    esac
+done
+
+# --
+
+if [ -z "$version" ] ; then
+    echo "ERROR: Flatcar SDK version to build container for cannot be autodetected and was not provided."
+    exit 1
+fi
+
+yell "\n######\n###### Building SDK container for version $version"
+
+if $checkout; then
+    if $dev; then
+        checkout_ebuild_repos main
+    else
+        checkout_ebuild_repos "$version"
+    fi
+fi
+
+create_versionfile "$version"
+
+if $existing; then
+    yell "Using existing container image"
+else
+    if [ -z "$tarball" ]; then
+        echo -e "\n###### Fetching SDK tarball ######"
+        tarball="flatcar-sdk-${arch}-${version}.tar.bz2"
+        if $dev; then
+            url="https://storage.googleapis.com/flatcar-jenkins/developer/sdk/${arch}/${version}/$tarball"
+        else
+            url="https://storage.googleapis.com/flatcar-jenkins/sdk/${arch}/${version}/$tarball"
+        fi
+
+        wget "$url"
+    else
+        if [ -f "$tarball" ]; then
+            version="$(echo "$tarball" | sed -n 's/.*-\([.0-9]\+\)\.tar\.bz2/\1/p')"
+            arch="$(echo "$tarball" | sed -n 's/flatcar-sdk-\(arm64\|amd64\)-[.0-9]\+\.tar\.bz2/\1/p')"
+            if [ -z "$version" -o -z "$arch" ]; then
+                echo "ERROR: Unable to determine version / arch from '$tarball'"
+                exit 1
+            fi
+        else
+            echo "ERROR: SDK tarball '$tarball' not found."
+            exit 1
+        fi
+    fi
+
+    yell "Importing SDK tarball"
+    docker import "$tarball" flatcar-sdk-tarball:${version}
+fi
+
+yell "Initialising the SDK container and building board packages"
+docker build -t flatcar-sdk-import:${version} \
+             --build-arg VERSION="${version}" \
+             -f sdk_lib/Dockerfile.sdk-container \
+             .
+
+for a in all arm64 amd64; do
+    yell "Creating '$a' arch SDK image"
+    rmarch=""
+    case $a in
+        arm64) rmarch="amd64-usr";;
+        amd64) rmarch="arm64-usr";;
+    esac
+    docker build -t "$sdk_container_common_registry/flatcar-sdk-${a}:${version}" \
```

Comment:
Out of curiosity, is there something preventing us to merge the following Dockerfiles: `sdk_lib/Dockerfile.sdk-container` and `sdk_lib/Dockerfile.lean-arch` ?

Reply:
I was thinking about pipelining / paralellising the tasks later : the 3 docker build commands producing the final images can run in parallel, while the first 2 (importing the tarball and installing the BSPs) cannot.

Reply:
Actually, let me add a check for whether the intermediate containers exist already and then skip the respective steps conditionally.